src/main/java/org/sonar/plugins/ldap/LdapContextFactory.java

```diff
@@ -97,8 +97,10 @@ public InitialDirContext createUserContext(String principal, String credentials)
   }
 
   private InitialDirContext createInitialDirContext(String principal, String credentials, boolean pooling) throws NamingException {
-    InitialLdapContext ctx = new InitialLdapContext(getEnvironment(principal, credentials, pooling), null);
+    final InitialLdapContext ctx;
     if (startTLS) {
+      ctx = new InitialLdapContext(getEnvironment(null, null, /* TODO(Godin): connection pooling requires proper TLS shutdown */false), null);
```

Comment:
Note for myself - http://docs.oracle.com/javase/7/docs/technotes/guides/jndi/jndi-ldap.html#POOL